src/visualisation/PlantVisualiser.cpp

```diff
@@ -0,0 +1,890 @@
+    {
+      BuildAttachmentMap();
+    }
+PlantVisualiser::~PlantVisualiser() = default;
+
+void PlantVisualiser::setPlant(const std::shared_ptr<MappedPlant>& plant) {
+  plant_ = plant;
+  BuildAttachmentMap();
+}
+
+void PlantVisualiser::ComputeGeometryForOrgan(int organId)
+{
+  auto result = plant_->getOrgans(-1, false);
+  auto organ_it = std::find_if(result.begin(), result.end(), [organId](const auto& o) {
+    return o->getId() == organId;
+  });
+  if(organ_it == result.end())
+  {
+    std::stringstream errMsg;
+    errMsg << "MappedPlant::ComputeGeometryForOrgan: organ not found: " << organId;
+    throw std::runtime_error(errMsg.str().c_str());
+  }
+  auto organ = *organ_it;
+  
+  if(organ->organType() == 4)
+  {
+    // 4 SHOULD mean leaf, so we do not check for successful cast
+		unsigned int point_space = 0, cell_space = 0;
+		
+		GenerateStemGeometry(organ, point_space, cell_space);
+		point_space += organ->getNumberOfNodes() * 3 * geometry_resolution_;
+		cell_space += (organ->getNumberOfNodes() - 1) * 2 * geometry_resolution_;
+		auto leaf = std::dynamic_pointer_cast<Leaf>(organ);
+		if(leaf->getLeafRandomParameter()->parametrisationType == 1)
+		{
+			GenerateRadialLeafGeometry(leaf, point_space, cell_space);
+			point_space += leaf->getNumberOfNodes() * 6 * 3;
+			cell_space += leaf->getNumberOfNodes() * 6;
+		}
+		else
+		{
+			int petiole_zone = 0;
+			for(int i = 0; i < leaf->getNumberOfNodes(); i++)
+			{
+				if(leaf->nodeLeafVis(leaf->getLength(i)))
+				{
+					petiole_zone = i;
+					break;
+				}
+			}
+			if(petiole_zone + 1 < leaf->getNumberOfNodes())
+			{
+				GenerateLeafGeometry(leaf, petiole_zone, point_space, cell_space);
+				point_space += (organ->getNumberOfNodes() - petiole_zone) * 6 * 3;
+				cell_space += (organ->getNumberOfNodes() - petiole_zone) * 6;
+			}
+		}
+  }
+  else
+  {
+    GenerateStemGeometry(organ);
+  }
+}
+
+void PlantVisualiser::ComputeGeometryForOrganType(int organType, bool clearFirst)
+{
+  auto organ_list = plant_->getOrgans(-1, false);
+		
+  // First we check if we have enough memory to support the geometry_
+  unsigned int point_space = 0;
+  unsigned int cell_space = 0;
+	unsigned int num_organs = 0;
+
+  for(auto organ : organ_list)
+  {
+    // Check against object, because organType can be -1
+		if(organ->organType() == organType || organType < 0)
+		{
+			if(organ->organType() == 4)
+			{
+				// 4 SHOULD mean leaf, so we do not check for successful cast
+				if(include_midline_in_leaf_)
+				{
+				  point_space += organ->getNumberOfNodes() * 3 * geometry_resolution_;
+				  cell_space += (organ->getNumberOfNodes() - 1) * 6 * geometry_resolution_;
+				}
+			  point_space += (organ->getNumberOfNodes()) * 4 * 3;
+				cell_space += ((organ->getNumberOfNodes()) - 1) * 4 * 3 + 40;
+			}
+			else
+			{
+				point_space += organ->getNumberOfNodes() * 3 * geometry_resolution_;
+				cell_space += (organ->getNumberOfNodes() - 1) * 6 * geometry_resolution_;
+			}
+		}
```

Comment:
This pre-spacing of the geometry buffers is to avoid any issues due to vector resizing as on some machines (specifically on mine) the resizing of the vectors too often during computation leads to issues.